google-cloud-spanner/src/test/java/com/google/cloud/spanner/spi/v1/GfeLatencyTest.java

```diff
@@ -0,0 +1,346 @@
+                                                    @Override
+                                                    public void sendHeaders(Metadata headers){
+                                                        headers.put(
+                                                                Metadata.Key.of("server-timing", Metadata.ASCII_STRING_MARSHALLER),
+                                                                String.format("gfet4t7; dur=%d", fakeServerTiming.get()));
+                                                        super.sendHeaders(headers);
+                                                    }
+                                                }, headers);
+                                    }
+                                })
+                        .build()
+                        .start();
+        optionsMap.put(SpannerRpc.Option.CHANNEL_HINT, 1L);
+        spanner = createSpannerOptions(address,server).getService();
+        databaseClient =
+                spanner.getDatabaseClient(DatabaseId.of(projectId, instanceId, databaseId));
+
+
+        mockSpannerNoHeader = new MockSpannerServiceImpl();
+        mockSpannerNoHeader.setAbortProbability(0.0D);
+        mockSpannerNoHeader.putStatementResult(MockSpannerServiceImpl.StatementResult.query(SELECT1AND2, SELECT1_RESULTSET));
+        mockSpannerNoHeader.putStatementResult(MockSpannerServiceImpl.StatementResult.update(UPDATE_FOO_STATEMENT, 1L));
+        addressNoHeader = new InetSocketAddress("localhost", 1025);
+        serverNoHeader =
+                NettyServerBuilder.forAddress(addressNoHeader)
+                        .addService(mockSpannerNoHeader)
+                        .build()
+                        .start();
+        spannerNoHeader = createSpannerOptions(addressNoHeader, serverNoHeader).getService();
+        databaseClientNoHeader =
+                spannerNoHeader.getDatabaseClient(DatabaseId.of(projectId, instanceId, databaseId));
+    }
+
+    @AfterClass
+    public static void stopServer() throws InterruptedException {
+        if (spanner != null) {
+            spanner.close();
+            server.shutdown();
+            server.awaitTermination();
+        }
+
+        if(spannerNoHeader != null){
+            spannerNoHeader.close();
+            serverNoHeader.shutdown();
+            serverNoHeader.awaitTermination();
+        }
+    }
+
+    @After
+    public void reset() {
+        mockSpanner.reset();
+        mockSpannerNoHeader.reset();
+    }
+
+    @Test
+    public void testGfeLatencyExecuteStreamingSql() throws InterruptedException {
+        try(ResultSet rs = databaseClient.singleUse().executeQuery(SELECT1AND2)){
+            rs.next();
+        }
+        Thread.sleep(WAIT_FOR_METRICS_TIME_MS);
+        long latency = getMetric(SpannerRpcViews.SPANNER_GFE_LATENCY_VIEW,"google.spanner.v1.Spanner/ExecuteStreamingSql");
+        assertThat(latency).isEqualTo(fakeServerTiming.get());
```

Comment:
Please don't use `assertThat`, use the default matchers `assertEquals`, `assertTrue`, ...